boule/triaxialellipsoid.py

```diff
@@ -0,0 +1,254 @@
+        """
+        Check if semimedium_axis is positive
+        """
+        if not value > 0:
+            raise ValueError(
+                f"Invalid semi-medium axis '{value}'. Should be greater than zero."
+            )
+        """
+        Check if semimedium axis is the middle of the 3 axes.
+        """
+        if self.semiminor_axis > value:
+            raise ValueError(
+                f"Invalid semi-minor / semi-medium axis combination.                 The semimedium axis must be larger than the semi-minor axis.                Semi-medium axis was '{value}' and the semi-minor axis was '{self.semiminor_axis}'"
+            )
+
+    @semiminor_axis.validator
+    def _check_semiminor_axis(
+        self, semiminor_axis, value
+    ):  # pylint: disable=no-self-use,unused-argument
+        """
+        Check if semiminor_axis is positive
+        """
+        if not value > 0:
+            raise ValueError(
+                f"Invalid semi-minor axis '{value}'. Should be greater than zero."
+            )
+
+    @geocentric_grav_const.validator
+    def _check_geocentric_grav_const(
+        self, geocentric_grav_const, value
+    ):  # pylint: disable=no-self-use,unused-argument
+        """
+        Warn if geocentric_grav_const is negative
+        """
+        if value < 0:
+            warn(f"The geocentric gravitational constant is negative: '{value}'")
+
+    @property
+    def a(self):
+        "The semimajor axis length [meters]"
+        return self.semimajor_axis
+
+    @property
+    def b(self):
+        "The semimedium axis length [meters]"
+        return self.semimedium_axis
+
+    @property
+    def c(self):
+        "The semiminor axis length [meters]"
+        return self.semiminor_axis
+
+    @property
+    def omega(self):
+        "The angular_velocity [rad s^-1]"
+        return self.angular_velocity
+
+    @property
+    def GM(self):
+        "The Geocentric Gravitational Constant [m^3 s^-2]"
+        return self.geocentric_grav_const
+
+    @property
+    def mean_radius(self):
+        """
+        The arithmetic mean radius :math:`R_1=(a+b+c)/3` [meters]
+        """
+        return (
+            1 / 3 * (self.semimajor_axis + self.semimedium_axis + self.semiminor_axis)
+        )
+
+    @property
+    def volume(self):
+        """
+        The volume of a triaxial ellipsoid :math: `V = /frac{4}{3} pi a b c ` [meters^3]
+        """
+        return (
+            4
+            * np.pi
+            / 3
+            * (self.semimajor_axis * self.semimedium_axis * self.semiminor_axis)
+        )
+
+    @property
+    def gravity_pole(self):
+        "The norm of the gravity vector on the ellipsoid at the poles [m/s²]"
+        raise NotImplementedError
+
+    def radius(self, latitude, longitude):
+        r"""
+        Return the radius of the Triaxial Ellipsoid at the given latitude and longitude [metres]
+        """
+        raise NotImplementedError
```

Comment:
We don't need to define a `radius` for the `TriaxialEllipsoid`. The `radius` is only suited for the `Sphere`.
```suggestion
```

Reply:
This method was mentioned in the feature request. 

I assumed this was a vector running from the centre of the ellipsoid (aka the origin [is it called the geocentric location??]) to a point on the surface at a given latitude and longitude. 
For a sphere this is a constant.
For a biaxial ellipsoid this maps to an ellipse along lines of longitude and a constant on lines of latitude.
For a triaxial ellipsoid I think it is this value (source: wikipedia page on ellipsoids).

![image](https://user-images.githubusercontent.com/29368602/98900592-10a48a80-24ed-11eb-9834-60d64a95d195.png)

Should this method be called something else?

Reply:
This is akin to the `geocentric_radius’s of `Ellipsoid` https://www.fatiando.org/boule/dev/api/generated/boule.Ellipsoid.html#boule.Ellipsoid.geocentric_radius

Reply:
Sorry for the misunderstanding: I though you were trying to implement a not implemented method in replacement of the `radius` attribute of the `Sphere`.
Now that I know what you meant by `radius`, I completely agree that the class should have this method.
Nevertheless, `radius` is a very misleading name. The `Ellipsoid` class has a [`geocentric_radius`](https://www.fatiando.org/boule/latest/api/generated/boule.Ellipsoid.html#boule.Ellipsoid.geocentric_radius) method which returns the quantity you describe. That name explicitly says that we are about to return the norm of the _radius_ vector to the surface of the ellipsoid measured from a geocentric coordinate system, so the origin is at the center of the ellipsoid.

In the `TriaxialEllipsoid`, the `geocentric_radius` should get the `longitude` and the `latitude`, as you already did. But, remember that on every Fatiando library we keep the following order for coordinates: `longitude`, `latitude`, `height`, ... or `easting`, `northing`, `height`, ..., except of very special cases where the reason is well documented.

So feel free to ignore this suggestion and please implement the `geocentric_radius` as you described.

Reply:
Thanks for clearing up the misunderstanding there.
I was completely wrong and didn't see geocentric_radius method.

I have removed the radius method from the latest commit.